src/Init/Data/Array/Basic.lean

```diff
@@ -487,6 +506,11 @@ def elem [BEq α] (a : α) (as : Array α) : Bool :=
 def toList (as : Array α) : List α :=
   as.foldr List.cons []
 
+/-- Prepends an `Array α` onto the front of a list.  Equivalent to `as.toList ++ l`. -/
+@[inline]
+def toListAppend (as : Array α) (l : List α) : List α :=
+  as.foldr List.cons l
+
```

Comment:
Just noting that while this one looks a little silly, it is quite useful in setting up tail-recursive versions of List operations in Std.